### ISSUE TITLE
Error fetching unavailablility of generation units with pandas client

### ISSUE THREAD

**Issue by SirWindfield** [EnergieID/entsoe-py#34]
I always get the following error when trying to fetch:

```
File "u:\code\new\entsoe-outage-fetcher.py", line 29, in <module>
    run()
  File "u:\code\new\entsoe-outage-fetcher.py", line 24, in run
    df = client.query_unavailability_of_generation_units(country_code, start=start_date, end=end_date, docstatus=None)
  File "C:\Users\me\AppData\Roaming\Python\Python36\site-packages\entsoe\entsoe.py", line 537, in year_wrapper
    in blocks]
  File "C:\Users\me\AppData\Roaming\Python\Python36\site-packages\entsoe\entsoe.py", line 536, in <listcomp>
    frames = [func(*args, start=_start, end=_end, **kwargs) for _start, _end
  File "C:\Users\me\AppData\Roaming\Python\Python36\site-packages\entsoe\entsoe.py", line 518, in pagination_wrapper
    df = func(*args, start=start, end=end, **kwargs)
  File "C:\Users\me\AppData\Roaming\Python\Python36\site-packages\entsoe\entsoe.py", line 792, in query_unavailability_of_generation_units
    df = parse_unavailabilities(content)
  File "C:\Users\me\AppData\Roaming\Python\Python36\site-packages\entsoe\parsers.py", line 352, in parse_unavailabilities
    with zipfile.ZipFile(BytesIO(response), 'r') as arc:
TypeError: a bytes-like object is required, not 'DataFrame'
```

`end_date`: Timestamp('2015-01-02 00:00:00+0100', tz='CET')
`start_date`: Timestamp('2015-01-01 00:00:00+0100', tz='CET')

`client.query_unavailability_of_generation_units('DE', start=start_date, end=end_date, docstatus=None)`

**EDIT:** I think you are not converting the zip file into a DataFrame, see [here](https://github.com/EnergieID/entsoe-py/blob/be6c724c4b9aac438954f79623d7ec0b51ab4848/entsoe/entsoe.py#L860). You do, however, seem to convert the bytes into a DataFrame [some lines below](https://github.com/EnergieID/entsoe-py/blob/be6c724c4b9aac438954f79623d7ec0b51ab4848/entsoe/entsoe.py#L895).

**EDIT 2:** Could you elaborate more on the difference between `generation` and `production` units? :)

**Comment by JrtPec** [comment]
1. I do not get the same error as you, so perhaps this has already been fixed (version 0.2.9).
2. This is indeed very confusing. The inherited calls obscure the return type of your request:

`EntsoePandasClient.query_unavailability_of_generation_units` calls its super: `EntsoeRawClient.query_unavailability_of_generation_units`, which says it returns bytes. But in fact, it calls `self.query_unavailability`, which in this case is `EntsoePandasClient.query_unavailability`, and returns a DataFrame.

Unavailability of Generation Units calls `doctype` A80.
Unavailability of Production Units calls `doctype` A77.
In the API docs these doctypes are described as "Generation Unavailability" and "Production Unavailability", so that doesn't help much. @jpaduart added this functionality in 559c7e9, maybe he can explain?

**Comment by JrtPec** [comment]
Update: #33 simplifies the code and makes it more transparent, so this bug should be fixed in version 0.2.10. If not, please re-open this issue.
